build/tools/Automation/Program.cs

```diff
@@ -2,180 +2,50 @@
-        ///
-        /// <param name="platformConstants">List of PlatformConstant containing platform release information</param>
-        /// <param name="yamlConstants">Deserialized Constants.ConstantsYaml which is ready for editing</param>
-        /// </Summary>
-        public abstract void UpdateConstants(List<PlatformConstant> platformConstants, List<Constant> yamlConstants);
-
-        /// <Summary>
-        /// Stores platform release information so it can be referenced when updating:
-        /// constants.yaml and versionsToBuild.txt files
-        /// </Summary>
-        public class PlatformConstant
-        {
-            /// <Summary>
-            /// The version of the platfom.
-            /// Example: 1.2.3
-            /// </Summary>
-            public string Version { get; set; } = string.Empty;
-
-            /// <Summary>
-            /// The sha of the platform's version.
-            /// Some platforms may not have a sha.
-            /// </Summary>
-            public string Sha { get; set; } = string.Empty;
-
-            /// <Summary>
-            /// The name of the platform.
-            /// Example: dotnet, golang, etc.
-            /// </Summary>
-            public string PlatformName { get; set; } = string.Empty;
-
-            /// <Summary>
-            /// The type of version that is being represented.
-            /// Example: sdk, aspnetcore, netcore, etc.
-            /// </Summary>
-            public string VersionType { get; set; } = string.Empty;
-
-            // TODO: Add fields for other feilds.
-            //  For example, python has GPG keys
-        }
-
-        /// <Summary>
-        /// This is used to deserialize Constants.ConstantsYaml file
-        /// </Summary>
-        public class Constant
-        {
-            public string Name { get; set; } = string.Empty;
-
-            public Dictionary<string, object> Constants { get; set; } = new Dictionary<string, object>();
-
-            public List<object> Outputs { get; set; } = new List<object>();
+            var serviceProvider = new ServiceCollection()
+                .AddScoped<IHttpClient, HttpClientImpl>()
+                .AddScoped<ILogger, LoggerImpl>()
+                .AddSingleton<IVersionService, VersionService>()
+                .AddSingleton<IYamlFileReaderService, YamlFileReaderService>()
+                .AddScoped<DotNet.DotNet>()
+                .AddLogging()
+                .BuildServiceProvider();
+
+            string oryxRootPath = args[1];
+            string platform = args[0].ToLower();
+            switch (platform)
+            {
+                case "dotnet":
+                    var dotNet = serviceProvider.GetRequiredService<DotNet.DotNet>();
+                    await dotNet.RunAsync(oryxRootPath);
+                    break;
+
+                default:
+                    Console.WriteLine($"Unsupported platform: {args[0]}");
```

Comment:
nit: we should use `platform` rather than `args[0]` here since it's a variable we already defined